server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/get/TransportGetRepositoriesAction.java

```diff
@@ -83,18 +84,33 @@ protected void masterOperation(
         if (repoNames.length == 0 || (repoNames.length == 1 && ("_all".equals(repoNames[0]) || "*".equals(repoNames[0])))) {
             return repositories.repositories();
         } else {
-            Set<String> repositoriesToGet = new LinkedHashSet<>(); // to keep insertion order
+            boolean seenWildcard = false;
+            final Set<String> repositoriesToGet = new LinkedHashSet<>(); // to keep insertion order
+            final Set<String> repositoriesNotToGet = new HashSet<>();
             for (String repositoryOrPattern : repoNames) {
-                if (Regex.isSimpleMatchPattern(repositoryOrPattern) == false) {
+                if (seenWildcard && repositoryOrPattern.length() > 1 && repositoryOrPattern.startsWith("-")) {
```

Comment:
I suppose we could iterate 2 times over the repositories, the first time to match repositories against include patterns ad the second to filter out matching repos with exclude patterns? Anyway the current code works too.

Reply:
Can we perhaps do this similar to the slm-change, where we first derive the include/exclude patterns and then run through the repos? And then extract the logic to a shared method to use such that we are sure that the include/exclude logic works the same.